app/controllers/application_controller.rb

```diff
@@ -16,6 +19,16 @@ def authenticate_request!
     render json: { errors: ['Not Authenticated'] }, status: :unauthorized
   end
 
+  def authenticate_request_admin!
+    unless admin_id_in_token? && auth_token[:type] == 'admin'
```

Comment:
Please move this  `auth_token[:type] == 'admin'`  to  `admin_id_in_token?` method